pyaedt/edb_grpc/core/edb_data/components_data.py

```diff
@@ -0,0 +1,962 @@
+import logging
+import re
+import warnings
+
+from pyaedt.edb_grpc.core.edb_data.padstacks_data import EDBPadstackInstance
+from pyaedt.generic.general_methods import is_ironpython
+
+if not is_ironpython:
+    try:
+        import numpy as np
+    except ImportError:
+        warnings.warn(
+            "The NumPy module is required to run some functionalities of EDB.\n"
+            "Install with \n\npip install numpy\n\nRequires CPython."
+        )
+from pyaedt.generic.general_methods import get_filename_without_extension
+from pyaedt.generic.general_methods import pyaedt_function_handler
+
+
+class EDBComponentDef(object):
+    """Manages EDB functionalities for component definitions.
+
+    Parameters
+    ----------
+    parent : :class:`pyaedt.edb_grpc.core.components.Components`
+        Inherited AEDT object.
+    comp_def : object
+        Edb ComponentDef Object
```

Comment:
Shouldn't the object be linked as the data type and the description be in plain language: EB component definition object.